keanu-project/src/main/java/io/improbable/keanu/vertices/NonProbabilistic.java

```diff
@@ -3,8 +3,8 @@
 public interface NonProbabilistic<T> extends Observable<T> {
 
     default boolean contradictsObservation() {
-        return isObserved() && getObservedValue().map(v -> !v.equals(calculate())).orElse(false);
+        return isObserved() && getObservedValue().map(v -> !v.equals(getObservedValue())).orElse(false);
```

Comment:
This will always succeed. Isn't it comparing the observed value to the observed value? It might be that calculate() can't be void....

Reply:
Aye....reverted the `calculate` change